src/huutonet/api.rs

```diff
@@ -2,15 +2,21 @@ use serde_json::Value;
 
 pub fn vahti_to_api(vahti: &str) -> String {
     let mut url = String::from("https://api.huuto.net/1.1/items?");
-    if vahti.contains('&') {
+    if vahti.contains('?') {
         // Easy parse
         url += &vahti[vahti.find('?').unwrap() + 1..];
     } else {
         // Difficult parse
         let mut args: Vec<&str> = vahti.split('/').collect();
         let args: Vec<&str> = args.drain(4..).collect();
+
+        let mut url_end = String::new();
         for arg in args.chunks_exact(2) {
-            url += &format!("&{}={}", arg[0], arg[1]);
+            url_end += &format!("&{}={}", arg[0], arg[1]);
```

Comment:
you can collect into a String.

You could map the chunks with the format!, as you do, and then just do .collect::<String>()